### DISCUSSION TITLE
react-aria Tabs State example - trying to understand what keyMap and getKeys are doing in state.collection

### DISCUSSION THREAD

**Discussion by BrendanBerkley** [adobe/react-spectrum#7043] [Q&A] [answered]
https://react-spectrum.adobe.com/react-aria/Tabs.html#state

If I take that basic code example and plug it into [Code Sandbox](https://codesandbox.io/p/sandbox/xenodochial-torvalds-grgmfh), and add `console.log(state?.collection);`, I get `{keyMap: Map(3), firstKey: 'home', lastKey: 'search', frozen: true}`.

But if

1. I search for keyMap in the react-aria docs nothing shows up, and
2. if I run this in a TS environment, it doesn't recognize keyMap but does suggest that getKeys is available, even though the latter never seems to output anything but an empty map.

Is there a gap in my understanding here? What's the correct way to get a list of all keys in a `TabListStateContext`?

**Comment by snowystinger** [comment]
Docs on Collections are still coming. For now, you'll need to rely on the source code.

What are you trying to do with the collection? Maybe we can help with that instead.

**Reply by BrendanBerkley** [reply, marked as answer]
I was having a few issues simultaneously, and I think I was fixating on the fact that I wasn't getting _anything_ that I expected, getKeys among them.

I fixed an unrelated issue and so this isn't a problem anymore. But I dug around for a bit and it seems like maybe there's just some issue on my end with types not matching reality. Probably outside of anyone else's purview.

![image](https://github.com/user-attachments/assets/4bb95e64-51f1-45b8-8a33-7dd8c4a7643b)

Thanks for your reply.
